simde/x86/sse4.2.h

```diff
@@ -37,6 +37,231 @@ SIMDE__BEGIN_DECLS
+  const int cmp_op = imm8 & 0x06;
+  const int polarity = imm8 & 0x30;
+  simde__m128i_private
+    bool_res_ = simde__m128i_to_private(simde_mm_setzero_si128()),
+    a_ = simde__m128i_to_private(a),
+    b_ = simde__m128i_to_private(b);
+  const int upper_bound = (128 / 8) - 1;
+  int a_invalid = 0;
+  int b_invalid = 0;
+  for(int i = 0 ; i < (upper_bound) ; i++) {
+    for(int j = 0; j< (upper_bound) ; j++){
+      int bitvalue = ((a_.i8[i] == b_.i8[j]) ? 1 : 0);
+      bool_res_.i8[i] |= (( bitvalue ) << j);
+      if(i == la)
+        a_invalid = 1;
+      if(j == lb)
+        b_invalid = 1;
+      switch(cmp_op){
+        case SIMDE_SIDD_CMP_EQUAL_ANY:
+          break;
+        case SIMDE_SIDD_CMP_RANGES:
+          break;
+        case SIMDE_SIDD_CMP_EQUAL_EACH:
+          if(a_invalid && b_invalid)
+            bool_res_.i8[i] |= (1 << j);
+          break;
+        case SIMDE_SIDD_CMP_EQUAL_ORDERED:
+          if(a_invalid && !b_invalid)
+            bool_res_.i8[i] |= (1 << j);
+          else if(a_invalid && b_invalid)
+            bool_res_.i8[i] |= (1 << j);
+          break;
+      }
+    }
+  }
+  int32_t int_res_1 = 0;
+  int32_t int_res_2 = 0;
+  switch(cmp_op) {
+    case SIMDE_SIDD_CMP_EQUAL_ANY:
+      for(int i = 0 ; i < (upper_bound) ; i++){
+        SIMDE__VECTORIZE_REDUCTION(|:int_res_1)
+        for(int j = 0 ; j < (upper_bound) ; j++){
+          int_res_1 |= (((bool_res_.i8[i] >> j) & 1) << i);
+        }
+      }
+      break;
+    case SIMDE_SIDD_CMP_RANGES:
+      for(int i = 0 ; i < (upper_bound) ; i++){
+        SIMDE__VECTORIZE_REDUCTION(|:int_res_1)
+        for(int j = 0 ; j < (upper_bound) ; j++){
+          int_res_1 |= ((((bool_res_.i8[i] >> j) & 1) & ((bool_res_.i8[i] >> (j + 1)) & 1)) << i);
+          j += 2;
+        }
+      }
+      break;
+    case SIMDE_SIDD_CMP_EQUAL_EACH:
+      for(int i = 0 ; i < (upper_bound) ; i++){
+        SIMDE__VECTORIZE_REDUCTION(|:int_res_1)
+        for(int j = 0 ; j < (upper_bound) ; j++){
+          int_res_1 |= (((bool_res_.i8[i] >> i) & 1) << i);
+        }
+      }
+      break;
+    case SIMDE_SIDD_CMP_EQUAL_ORDERED:
+      int_res_1 = (imm8 & 1) ? 0xff : 0xffff;
+      for(int i = 0 ; i < (upper_bound) ; i++){
+        int k = i;
+        SIMDE__VECTORIZE_REDUCTION(|:int_res_1)
+        for(int j = 0 ; j < (upper_bound-i) ; j++){
+          int_res_1 &=  (((bool_res_.i8[k] >> j) & 1 ) << i) ;
+          k += 1;
+        }
+      }
+  }
+  for(int i = 0; i < (upper_bound) ; i++){
+    if(polarity & 1){
+      if((polarity >> 1) & 1) {
+        if (i >= lb) {
+          int_res_2 |= (((int_res_1 >> i) & 1) << i);
+        }
+        else {
+          int_res_2 |= ((((int_res_1 >> i) & 1) ^ (-1)) << i);
+        }
+      }
+      else{
+        int_res_2 |= ((((int_res_1 >> i) & 1) ^ (-1)) << i);
+      }
+    }
+    else{
+      int_res_2 |= ( ((int_res_1 >> i) & 1) << i);
+    }
+  }
+  return ( (int_res_2 == 0) & (lb > upper_bound) );
```

Comment:
Extra spaces inside the parenthesis, and you don't need the outer parenthesis.  Just `return (int_res_2 == 0) & (lb > upper_bound);` is fine.  Or even `return !int_res_2 & (lb > upper_bound);`

Reply:
corrected this